test/jdk/com/sun/java/swing/plaf/gtk/bug6492108.java

```diff
@@ -0,0 +1,155 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.GridLayout;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.awt.image.BufferedImage;
+
+import javax.swing.JEditorPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTextArea;
+import javax.swing.JTextPane;
+import javax.swing.UIManager;
+import javax.swing.text.JTextComponent;
+
+import jtreg.SkippedException;
+
+/*
+ * @test
+ * @bug 6492108 8160755
+ * @key headful
+ * @requires (os.family == "linux")
+ * @summary Verifies that the background is painted the same for
+ *          JTextArea, JTextPane, and JEditorPane.
+ * @library /javax/swing/regtesthelpers /test/lib
+ * @build SwingTestHelper
+ * @run main/othervm bug6492108
+ */
+
+public class bug6492108 extends SwingTestHelper {
+
+    private JPanel panel;
+
+    public static void main(String[] args) throws Throwable {
+        try {
+            UIManager.setLookAndFeel(
+                "com.sun.java.swing.plaf.gtk.GTKLookAndFeel");
+        } catch (Exception e) {
+            throw new SkippedException("GTK LAF is not supported on this system;" +
+                    " test passes");
+        }
+        new bug6492108().run(args);
+    }
+
+    private static void addTextComps(Container parent,
+                                     Class<? extends JTextComponent> type)
+            throws Throwable
+    {
+        JTextComponent text = type.newInstance();
+        addTextComp(parent, text);
+
+        text = type.newInstance();
+        text.setEditable(false);
+        addTextComp(parent, text);
+
+        text = type.newInstance();
+        text.setEnabled(false);
+        addTextComp(parent, text);
+
+        text = type.newInstance();
+        text.setEnabled(false);
+        text.setEditable(false);
+        addTextComp(parent, text);
+    }
+
+    private static void addTextComp(Container parent, JTextComponent text) {
+        JScrollPane sp = new JScrollPane(text);
+        text.setFocusable(false); // to avoid showing the blinking caret
+        sp.setPreferredSize(new Dimension(150, 150));
+        sp.setHorizontalScrollBarPolicy(JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+        sp.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_NEVER);
+        parent.add(sp);
+    }
+
+    protected Component createContentPane() {
+        panel = new JPanel();
+        panel.setLayout(new GridLayout(3, 4));
+        try {
+            addTextComps(panel, JTextArea.class);
+            addTextComps(panel, JEditorPane.class);
+            addTextComps(panel, JTextPane.class);
+        } catch (Throwable t) {
+            fail("Problem creating text components");
+        }
+        return panel;
+    }
+
+    private void onEDT10() {
+        requestAndWaitForFocus(panel);
+    }
+
+    private void onBackgroundThread20() {
```

Comment:
This method should run on EDT because it accesses the state of the components.

Reply:
Updated to run this on EDT.